src/govuk/components/skip-link/skip-link.js

```diff
@@ -0,0 +1,81 @@
+function SkipLink ($module) {
+  this.$module = $module
+}
+
+/**
+ * Initialise the component
+ */
+SkipLink.prototype.init = function () {
+  var $module = this.$module
+  // Check for module
+  if (!$module) {
+    return
+  }
+
+  $module.addEventListener('click', this.handleClick.bind(this))
+}
+
+/**
+* Click event handler
+*
+* @param {MouseEvent} event - Click event
+*/
+SkipLink.prototype.handleClick = function (event) {
+  var target = event.target
+
+  if (this.focusTarget(target)) {
+    event.preventDefault()
+  }
+}
+
+/**
+ * Focus the target element
+ *
+ * @param {HTMLElement} $target - Event target
+ * @returns {boolean} True if the target was able to be focussed
+ */
+SkipLink.prototype.focusTarget = function ($target) {
+  // If the element that was clicked does not have a href, return early
+  if ($target.href === false) {
+    return false
+  }
+
+  var contentId = this.getFragmentFromUrl($target.href)
+  var $content = document.getElementById(contentId)
+  if (!$content) {
+    return false
+  }
+
+  if (!$content.getAttribute('tabindex')) {
```

Comment:
To make this easier to follow, what about making the variable names consistent with the method name and refer to the thing we want to focus as the `target`, rather than `content`?

```suggestion
  var targetId = this.getFragmentFromUrl($target.href)
  var $target = document.getElementById(contentId)
  if (!$target) {
    return false
  }

  if (!$target.getAttribute('tabindex')) {
```

etc…

Reply:
I tried to find a better name for the variable but we're constrained by the existing naming conventions here. `$target` is already defined, it contains the skip link. This function duplicates logic in the error summary (ideally we should be extracting this to be a reusable function but as just discussed in a different context doing that will be part a bigger piece of work) so I'm loath to rename the function or its parameter in order to keep that linkage. 

In the error summary we call the elements to focus by the HMTL element name (`$input` and `$legendOrLabel`) but `$main` in this context felt too specific, so I settled on `$content`.

Reply:
That's a good point – but I think points to a bigger naming issue (which is absolutely already present in the error summary) and which is potentially even more confusing… `target` means two different things in two different parts of the code. Within the context of `handleClick` target refers to the 'event target' (the thing that was clicked), but `focusTarget` at least suggests that the target is the element that we want to focus.

Base on the function's signature (`focusTarget($target)`) you would definitely expect it to focus whatever `$target` is. And that's also what the doc block says it does! But it **doesn't** do that – it focuses whatever `$target` links to. The target of the `$target`, I guess…?

I think this is worth trying to fix rather than introducing the issue to another component.

Reply:
You're right, it could be quite confusing. I've just tried to improve the wording just for this component, have a look.